airbyte-integrations/connectors/source-instagram/source_instagram/client/api.py

```diff
@@ -0,0 +1,268 @@
+            }
+
+    def _params(self) -> Dict:
+        return {"metric": self.LIFETIME_METRICS, "period": self.period}
+
+    def _get_insight_records(self, params) -> Iterator[Any]:
+        return self._api.account.get_insights(params=params)
+
+
+class UserInsightsAPI(IncrementalStreamAPI):
+    METRICS_BY_PERIOD = {
+        "day": [
+            "email_contacts",
+            "follower_count",
+            "get_directions_clicks",
+            "impressions",
+            "phone_call_clicks",
+            "profile_views",
+            "reach",
+            "text_message_clicks",
+            "website_clicks",
+        ],
+        "week": ["impressions", "reach"],
+        "days_28": ["impressions", "reach"],
+        "lifetime": ["online_followers"],
+    }
+
+    state_pk = "date"
+
+    buffer_days = 29
+    days_increment = 1
+
+    def __init__(self, api):
+        super().__init__(api=api)
+        self._state = api._start_date
+
+    def list(self, fields: Sequence[str] = None) -> Iterator[dict]:
+        account_id = self._api.account.get("id")
+        for params_per_day in self._params():
+            insight_list = []
+            for params in params_per_day:
+                insight_list += self._get_insight_records(params=params)
+            if not insight_list:
+                continue
+
+            insight_record = {"id": account_id}
+            for insight in insight_list:
+                key = (
+                    f"{insight.get('name')}_{insight.get('period')}"
+                    if insight.get("period") in ["week", "days_28"]
+                    else insight.get("name")
+                )
+                insight_record[key] = insight.get("values")[0]["value"]
+                if not insight_record.get("date"):
+                    insight_record["date"] = insight.get("values")[0]["end_time"]
+            yield self.state_filter(insight_record)
+
+    def _params(self) -> Iterator[List]:
+        buffered_start_date = max(self._state.add(minutes=1), pendulum.now().subtract(days=self.buffer_days))
+        end_date = pendulum.now()
+
+        while buffered_start_date <= end_date:
+            params_list = []
+            for period, metrics in self.METRICS_BY_PERIOD.items():
+                params_list.append(
+                    {
+                        "metric": metrics,
+                        "period": [period],
+                        "since": buffered_start_date.to_datetime_string(),
+                        "until": buffered_start_date.add(days=self.days_increment).to_datetime_string(),
+                    }
+                )
+            yield params_list
+            buffered_start_date = buffered_start_date.add(days=self.days_increment)
+
+    def _get_insight_records(self, params):
+        return self._api.account.get_insights(params=params)._queue
+
+
+class MediaAPI(StreamAPI):
+    INVALID_CHILDREN_FIELDS = ["caption", "comments_count", "is_comment_enabled", "like_count", "children"]
```

Comment:
why are these invalid? can you link to the relevant documentation?

Reply:
I didn't find this in the documentation, but in practice it is. Children objects can only be of the `media_type == "CAROUSEL_ALBUM"`.
And children object does not support `INVALID_CHILDREN_FIELDS` fields, so I exclude them when I try to get child objects to avoid an error.

Reply:
👍🏼

Reply:
Can you leave this explanation as a comment in he code?